app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -1564,6 +1690,65 @@ export function testsForRpcMethodSupportingBlockParam(
       });
     });
 
+    for (const paramIndex of [...Array(numberOfParameters).keys()]) {
+      if (paramIndex === blockParamIndex) {
+        // testing changes in block param is covered under later tests
+        continue;
+      }
```

Comment:
I'm not understanding why we want to skip these cases for this case?

Reply:
These tests are basically meant to ensure that the middleware correctly recognizes any parameter changes, and doesn't reuse a cache for different params. We want to check this for the block parameter as well, but, that one isn't so simple to make an arbitrary mutation to. The change will have other side-effects because the value is used by the caching middleware in other ways (it determines which block cache to use).

We do want to test the case where the block reference changes too, but, this test is within the block:

```
describe.each([
  ['given no block tag', undefined],
  ['given a block tag of "latest"', 'latest'],
])`
```

i.e. the block reference is always set to a static value in this test, that's what this category is for.

The case where the block reference is changed is covered by the test "does not reuse the result of a previous request if it was made with different arguments than this one", about ~1300 lines below here.